src/ui/components/ProtocolViewer.tsx

```diff
@@ -0,0 +1,91 @@
+import classNames from "classnames";
+import { padStart } from "lodash";
+import { useSelector } from "react-redux";
+import { getProtocolRequests, RequestSummary } from "ui/reducers/protocolMessages";
+
+const msAsMinutes = (ms: number) => {
+  const seconds = Math.round(ms / 1000.0);
+  return `${Math.floor(seconds / 60)}:${padStart(String(seconds % 60), 2, "0")}`;
+};
+
+const fullMethod = (request: RequestSummary): string => {
+  return `${request.class}.${request.method}`;
+};
+
+type RequestSummaryChunk = {
+  count: number;
+  method: string;
+  pending: boolean;
+  errored: boolean;
+  startedAt: number;
+};
+
+type ChunkReducer = {
+  chunks: RequestSummaryChunk[];
+  current: RequestSummaryChunk;
+};
+
+const chunkedRequests = (requests: RequestSummary[]): RequestSummaryChunk[] => {
+  return requests.reduce(
+    (acc: ChunkReducer, request: RequestSummary) => {
+      const { current } = acc;
+      if (
+        current.method === fullMethod(request) &&
+        current.pending === request.pending &&
+        current.errored === request.errored
+      ) {
+        current.count++;
+      } else {
+        acc.chunks.push(current);
+        acc.current = {
+          count: 1,
+          errored: request.errored,
+          method: fullMethod(request),
+          pending: request.pending,
+          startedAt: request.recordedAt,
+        };
+      }
+      return acc;
+    },
+    {
+      chunks: [],
+      current: {
+        count: 0,
+        method: "",
+        pending: false,
+        errored: false,
+        startedAt: 0,
+      },
+    }
+  ).chunks;
+};
+
+const ProtocolViewer = () => {
+  const requests = useSelector(getProtocolRequests);
+  const chunks = chunkedRequests(requests);
```

Comment:
:speech_balloon: was debating if it's worth memoizing this, but I guess this will only re-render when that list changes anyway

Reply:
yep, I had the same thought basically